File: src/test/scala/mrtjp/fengine/testimpl/PortlessWireTileImpl.scala
package mrtjp.fengine.testimpl

import mrtjp.fengine.TileCoord._
import mrtjp.fengine.tiles.{FEPortlessWireTile, FETile}

class PortlessWireTileImpl(override val connMask:Int) extends FETile with FEPortlessWireTile with TTestFETile
{
    override def getStringRenderRows:Seq[String] = {
        val cN = if ((connMask&bitNorth) != 0) "║" else " "
        val cE = if ((connMask&bitEast) != 0) "═" else " "
        val cS = if ((connMask&bitSouth) != 0) "║" else " "
        val cW = if ((connMask&bitWest) != 0) "═" else " "
        val cC = "╳╨╥║╡╝╗╣╞╚╔╠═╩╦╬".apply(connMask>>2)

        Seq(
            s"╭   $cN   ╮",
            s"    $cN    ",
            s"$cW$cW$cW$cW$cC$cE$cE$cE$cE",
            s"    $cS    ",
            s"╰   $cS   ╯"
        )
    }
}

File: src/compiletest/java/mrtjp/fengine/testimpl/PortlessWireTileImpl.java
package mrtjp.fengine.testimpl;

import mrtjp.fengine.api.IPathFinder;
import mrtjp.fengine.api.IPathFinderManifest;
import mrtjp.fengine.tiles.FEPortlessWireTile;

import java.util.HashSet;
import java.util.Set;

public class PortlessWireTileImpl extends FEPortlessWireTile {

    private final int connMask;

    public final Set<Integer> inputRegisters = new HashSet<>();

    public PortlessWireTileImpl(int connMask) {
        this.connMask = connMask;
    }

    @Override
    public void locate(IPathFinder pathFinder) {
        inputRegisters.clear();
    }

    @Override
    public void searchManifest(IPathFinderManifest manifest) {
        inputRegisters.addAll(manifest.getOutputRegisters()); // Found output registers feed into this wire as inputs
    }

    //@formatter:off
    @Override protected int getConnMask() { return connMask; }
    //@formatter:on
}
